packages/types/src/message-metadata.ts

```diff
@@ -8,15 +11,302 @@ export interface MessageMetadata {
    * @description A human readable alphanumeric string representing your application's metadata event.
    * The value of this field may appear in the UI to developers.
    */
-  event_type: string;
+  event_type?: string;
   /**
    * @description A free-form object containing whatever data your application wishes to attach to messages.
    */
-  event_payload: {
+  event_payload?: {
     [key: string]: string | number | boolean | MessageMetadataEventPayloadObject | MessageMetadataEventPayloadObject[];
   };
+
+  /**
+   * @description An array of work object entities.
+   */
+  entities?: EntityMetadata[];
 }
 
+// ------------------------------
+// Event metadata
+// ------------------------------
+
+export interface EventMessageMetadata extends Omit<MessageMetadata, 'entities'> {}
+
 export interface MessageMetadataEventPayloadObject {
   [key: string]: string | number | boolean;
 }
+
+export interface MessageMetadataEventPayloadAttributes {
+  [key: string]: string | number | boolean | MessageMetadataEventPayloadAttributes;
+}
+
+// ------------------------------
+// Work object metadata
+// ------------------------------
+
+/**
+ * @description Metadata that represents a work object entity.
+ */
+export interface EntityMetadata {
+  /**
+   * @description Entity type.
+   */
+  entity_type: EntityType | string;
+  /**
+   * @description Schema for the given entity type.
+   */
+  entity_payload: {
+    attributes: EntityAttributes;
+    fields?: FileEntityFields | TaskEntityFields | IncidentEntityFields | ContentItemEntityFields;
```

Comment:
⭐ praise: I'm a fan of this union implementation with the enum for `entity_type` above!

Reply:
> does "slack#/entities/item" require a fields type?

It does not 🙂 'item' is the generic type so we don't define any common fields